#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: Workaround nptl fix to reflect pthread attribute.
# DP: Related bugs: #266507: NPTL (0.60) quirks with pthread_create (ignores attributes)
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
# DP: Patch author: 
# DP: Upstream status: In CVS
# DP: Status Details: For only sarge.
# DP: Date: 2004-09-22

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- nptl/pthread_create.c~	2003-09-09 03:17:15.000000000 -0400
+++ nptl/pthread_create.c	2004-08-17 21:48:47.000000000 -0400
@@ -399,7 +399,7 @@
 
   /* Determine scheduling parameters for the thread.
      XXX How to determine whether scheduling handling is needed?  */
-  if (0 && attr != NULL)
+  if (attr != NULL)
     {
       if (iattr->flags & ATTR_FLAG_NOTINHERITSCHED)
 	{

